test/secrets_manager_example_test.go

```diff
@@ -0,0 +1,116 @@
+	// A unique ID we can use to namespace resources so we don't clash with anything already in the AWS account or
+	// tests running in parallel
+	uniqueID := random.UniqueId()
+	secretKey := fmt.Sprint("secret_key_", uniqueID)
+	secretValue := fmt.Sprint("secret_value_", uniqueID)
+	// Construct the terraform options with default retryable errors to handle the most common retryable errors in
+	// terraform testing.
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		// The path to where our Terraform code is located
+		TerraformDir: workingDir,
+
+		// Variables to pass to our Terraform code using -var options
+		Vars: map[string]interface{}{
+			"random_id":    uniqueID,
+			"region":       awsRegion,
+			"secret_key":   secretKey,
+			"secret_value": secretValue,
+		},
+	})
+
+	// Save the Terraform Options struct, instance name, and instance text so future test stages can use it
+	test_structure.SaveTerraformOptions(t, workingDir, terraformOptions)
+
+	// This will run `terraform init` and `terraform apply` and fail the test if there are any errors
+	terraform.InitAndApply(t, terraformOptions)
+}
+
+// Validate that the Secret created contains the values as expected
+func validateSecretsContainSecrets(t *testing.T, workingDir string) {
+	// Load the Terraform Options saved by the earlier deploy_terraform stage
+	terraformOptions := test_structure.LoadTerraformOptions(t, workingDir)
+
+	awsRegion := test_structure.LoadString(t, workingDir, "awsRegion")
+
+	// Check that two secrets were created
+	secret_arns := terraform.OutputList(t, terraformOptions, "secret_arns")
+	assert.Len(t, secret_arns, 2)
+
+	// Assert that the two secret names are correct
+	names := terraform.OutputList(t, terraformOptions, "secret_names")
+	assert.Len(t, names, 2)
+
+	secretKey := terraformOptions.Vars["secret_key"].(string)
+	secretValue := terraformOptions.Vars["secret_value"].(string)
+
+	// Check that each of the secrets can be retrieved
+	for _, arn := range secret_arns {
+		var secret map[string]string
+		secretStr := aws.GetSecretValue(t, awsRegion, arn)
+		json.Unmarshal([]byte(secretStr), &secret)
+
+		// Check that the keys all exist in the secret
+		assert.Len(t, secret, 2, "Secret does not contain %d keys. got: %d", 2, len(secret))
+
+		// Check that the keys have the expected prefix
+		for k := range secret {
+			assert.True(t, strings.HasPrefix(k, secretKey), "Secret key %s does not have prefix %s", k, secretKey)
+
+			value := secret[k]
+			assert.True(t, strings.HasPrefix(value, secretValue), "Secret value %s does not have prefix %s", value, secretValue)
+		}
+	}
+
+	// Run `terraform output` to get the value of an output variable
+	secret_arn_references := terraform.OutputList(t, terraformOptions, "secret_arn_references")
+
+	// Ensure the secret_arn_references format is as expected
+	assert.Len(t, secret_arn_references, 4, "Expected %d secret_arn_references, got: %d", 4, len(secret_arn_references))
```

Comment:
@JosueMolinaMorales Could you also add a check for the format? The format should be
`secret_arn:secret_key::`